torch/distributed/device_mesh.py

```diff
@@ -203,6 +203,8 @@ def __init__(
                 if isinstance(mesh, torch.Tensor)
                 else torch.tensor(mesh, dtype=torch.int)
             )
+            if mesh.device.type != "cpu":
```

Comment:
No strong opinion, but I think we could also do this:
```
            self.mesh = (
                mesh.detach().cpu()
                if isinstance(mesh, torch.Tensor)
                else torch.tensor(mesh, dtype=torch.int)
            )
```

We want to avoid recurring host/device syncs but sync on initialization is okay.